Library/Formula/mingw.rb

```diff
@@ -0,0 +1,123 @@
+class Mingw < Formula
+  def arch
+    if Hardware::CPU.type == :intel
+      if MacOS.prefer_64_bit?
+        "x86_64"
+      else
+        "i686"
+      end
+    elsif Hardware::CPU.type == :ppc
+      if MacOS.prefer_64_bit?
+        "powerpc64"
+      else
+        "powerpc"
+      end
+    end
+  end
+
+  def osmajor
+    `uname -r`.chomp
+  end
+
+  desc "Minimalist GNU for Windows"
+  homepage "http://www.mingw.org/"
+  url "http://downloads.sourceforge.net/project/mingw-w64/mingw-w64/mingw-w64-release/mingw-w64-v3.1.0.tar.bz2"
+  mirror "http://sourceforge.net/projects/mingw-w64/files/mingw-w64/mingw-w64-release/mingw-w64-v3.1.0.tar.bz2/download?use_mirror=kent"
+  sha256 "ece7a7e7e1ab5e25d5ce469f8e4de7223696146fffa71c16e2a9b017d0e017d2"
+
+  head "svn://gcc.gnu.org/svn/gcc/trunk"
+
+  resource "gcc48" do
+    url "http://ftpmirror.gnu.org/gcc/gcc-4.8.5/gcc-4.8.5.tar.bz2"
+    sha256 "22fb1e7e0f68a63cee631d85b20461d1ea6bda162f03096350e38c8d427ecf23"
+  end
+
+
+  #option "with-java", "Build the gcj compiler"
+  #option "with-all-languages", "Enable all compilers and languages, except Ada"
+  # option "with-nls", "Build with native language support (localization)"
+  #option "with-jit", "Build the jit compiler"
+  #option "without-fortran", "Build without the gfortran compiler"
+  # enabling multilib on a host that can't run 64-bit results in build failures
+  #option "without-multilib", "Build without multilib support" if MacOS.prefer_64_bit?
+
+  depends_on "gmp4"
+  depends_on "gcc48" => ["with-all-languages"]
+  depends_on "binutils"
+  depends_on "isl011"
+  depends_on "isl012"
+  depends_on "cloog018"
```

Comment:
They'll need to be vendored in this formula or this submitted to homebrew-versions instead.